### ISSUE TITLE
cuda in transformers backend?

### ISSUE THREAD

**Issue by kno10** [mudler/LocalAI#1344]
The `transformers` backend does not appear to use the GPU.
I could not get any of the settings such as `cuda: true` or `device: cuda` to work. In the code, I could not see any of these options being used in the backend, either, yet.
Maybe copy the relevant lines from the `diffusers` backend?
https://github.com/mudler/LocalAI/blob/9482acfdfc502fcbf29efa7a4feda045d768f95a/backend/python/diffusers/backend_diffusers.py#L223-L233

**Comment by mudler** [comment]
> The `transformers` backend does not appear to use the GPU. I could not get any of the settings such as `cuda: true` or `device: cuda` to work. In the code, I could not see any of these options being used in the backend, either, yet. Maybe copy the relevant lines from the `diffusers` backend?
> 
> https://github.com/mudler/LocalAI/blob/9482acfdfc502fcbf29efa7a4feda045d768f95a/backend/python/diffusers/backend_diffusers.py#L223-L233

good point - it was not hooked as I could not test it as well. please feel free to take a stab at it

**Comment by LeonSijiaLu** [comment]
I think this issue has been addressed https://github.com/mudler/LocalAI/blob/master/backend/python/transformers/transformers_server.py#L91, looks like we can close it, but please correct me if I missed something @mudler

**Comment by Dennisadira** [comment]
This appears to be implemented now — CUDA detection and routing are wired into the current `transformers` backend:

- `backend/python/transformers/backend.py:74` — `self.CUDA = torch.cuda.is_available()`
- `backend/python/transformers/backend.py:111-116` — when CUDA is available, `device_map` is set to `request.MainGPU` or `cuda:0`
- `backend/python/transformers/backend.py:272-273` — input tensors are moved to `cuda` for inference

@LeonSijiaLu's [comment from May 2024](https://github.com/mudler/LocalAI/issues/1344#issuecomment-2089486459) noted this had landed; the code paths are still present on master. Probably safe to close. /cc @mudler
